### PR TITLE
Add Summary class

### PR DESCRIPTION
**Pull request by ryantianj**
Add summary class and its test cases.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -238,7 +238,75 @@ _{more aspects and alternatives to be added}_
...
+![Sequence Diagram for Review](images/ReviewSequenceDiagram.png)
+
+The following activity diagram shows how the review operation works:
+
+![Sequence Diagram for Review](images/ReviewActivityDiagram.png)
+
+
+Given below is an example usage scenario of adding a Review:
+
+Step 1. The user adds a new Review using the `add` command e.g. `add n/test e/test@gmail.com p/12344321 a/test c/att rv/test review`
```
Comment (marcuschj) [inline docs/DeveloperGuide.md:266]: Maybe you could give some details on how the `add` command is parsed by the `Parser`?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -124,7 +124,7 @@ public static Review parseReview(String review) throws ParseException {
             throw new ParseException(Review.MESSAGE_CONSTRAINTS);
         }
         if (trimmedReview.equals("")) {
-            trimmedReview = "- No Review -";
+            trimmedReview = Review.EMPTY_REVIEW;
```
Comment (marcuschj) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:127]: Nice use of abstraction here

File: src/main/java/seedu/address/model/summary/Summary.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    /**
+     * Sets the total number of contacts in the addressBook.
+     * @return size of AddressBook.
+     */
+    private int setNumberOfContacts() {
+        return personList.size();
+    }
+
+    private int setPercentageRatings(Model model) {
```
Comment (marcuschj) [inline src/main/java/seedu/address/model/summary/Summary.java:44]: What does this function do? For now it returns a placeholder value of 0, am I right

Comment (ryantianj) [reply src/main/java/seedu/address/model/summary/Summary.java:44]: Yes, it returns a 0 for now. I'm awaiting changes done to Rate command/class before implementing this function

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 2 above)
Comment (ryantianj) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:127]: Thanks

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ryantianj) [reply docs/DeveloperGuide.md:266]: hmmm I felt that this was convered in the sequence diagram

**Pull request merged**